src/foam/nanos/ruler/RuledDAO.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.ruler',
+  name: 'RuledDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: 'Finds applicable ruled (i.e, rule-like) object by FindRuledCommand.',
+
+  javaImports: [
+    'foam.dao.ArraySink',
+    'static foam.mlang.MLang.*'
+  ],
+
+  methods: [
+    {
+      name: 'cmd_',
+      javaCode: `
+        if ( obj instanceof FindRuledCommand ) {
+          var sink = (ArraySink) getDelegate()
+            .where(AND(
+              INSTANCE_OF(Ruled.getOwnClassInfo()),
```

Comment:
I don't think this is needed. It will just slow things down and it should always be the case anyway.

Reply:
This is to safety check in case RuledDAO was added to a non-Ruled model/DAO.

Reply:
Better to do that just once then when we setOf(), rather than on every object in every select.

Reply:
Fixed in https://github.com/foam-framework/foam2/pull/4484/commits/43e617028b5148a6aab83bdae951667221c5c31d.